api/src/main/java/javax/enterprise/context/spi/ActiveContext.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * JBoss, Home of Professional Open Source
+ * Copyright 2013, Red Hat, Inc., and individual contributors
+ * by the @authors tag. See the copyright.txt in the distribution for a
+ * full listing of individual contributors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package javax.enterprise.context.spi;
+
+import javax.enterprise.inject.Instance;
+import java.lang.annotation.Annotation;
+
+/**
+ * Representation of a context which has been activated
+ *
+ * @author John D. Ament
+ */
+public interface ActiveContext {
+
+    /**
+     * Provides access to an internal ID used to uniquely identify this active context
+     *
+     * @return the internal ID of this context, if it is supported
+     */
+    String getId();
+
+    /**
+     * Associates the context to the given thread.
+     * @param thread the thread to associate with.  This does not remove associations with other threads, but acts additive
+     * @throws IllegalArgumentException if the given thread is already associated with this context.
+     */
+    void associate(Thread thread);
```

Comment:
@tremes No, its not started in the current thread.  One of the ideas here is to dis-associate the threading model and the context model.  We can still map back to the threading model via this method, but ideally the context that gets created is managed outside of any thread, and injection points are satisfied from the owning context.